inst/pages/multimedia.qmd

```diff
@@ -0,0 +1,590 @@
+  theme_classic() +
+  labs(
+    x = "Overall Indirect Effect",
+    y = "Frequency",
+    title = "Bootstrap Distribution of Overall Indirect Effect (B = 100)"
+  )
+
+```
+
+Finally, we visualize the **mediator-specific indirect effects** using a forest 
+plot, displaying point estimates and 95% CIs to highlight the most influential 
+mediators.
+
+```{r visualization-for-mediation-specific-indirect-effects}
+
+###############################################
+# Bootstrap mediator-specific indirect effect # 
+###############################################
+
+# Define an mediation-specific indirect effect function manually
+indirect_each <- function(mdl, exper) {
+  res <- estimate(mdl, exper)
+  alpha <- sapply(res@mediation@estimates, function(m) coef(m)["treatmentT1"])
+  names(alpha) <- sub("\\.treatmentT1$", "", names(alpha))
+  beta <- coef(res@outcome@estimates$dysbiosis)[names(alpha)]
+  indirect <- alpha * beta
+  names(indirect) <- names(alpha)
+  indirect <- indirect[!is.na(indirect)]
+  return(indirect)
+}
+
+set.seed(12345)
+boot_each <- bootstrap(mdl, exper, c(indirect = indirect_each), B = 100)
+summary(boot_each$indirect)
+
+# Summarize the bootstrap results
+lower_upper <- apply(boot_each$indirect, 2, function(x) quantile(x, c(0.025, 0.975), na.rm=TRUE))
+means <- apply(boot_each$indirect, 2, mean, na.rm=TRUE)
+
+# Convert to data frame
+summary_df <- data.frame(
+  mediator = colnames(boot_each$indirect),
+  estimate = means,
+  lower = lower_upper[1,],
+  upper = lower_upper[2,]
+)
+summary_df <- summary_df[-1, ]
+
+# Mark significance if CI does not cross zero
+summary_df$significant <- with(summary_df, lower > 0 | upper < 0)
+
+# Remove the species with 0 confidence interval length
+summary_df <- summary_df %>% 
+  filter((upper - lower) != 0)
+
+# Add rankings
+summary_df$rank <- ifelse(summary_df$significant, 1, 2)  # 1 for significant, 2 for not
+summary_df <- summary_df[order(summary_df$rank, -abs(summary_df$estimate)), ]
+
+# Forest plot
+ggplot(summary_df, aes(x=estimate, y=reorder(mediator, estimate), color=significant)) +
+  geom_point() +
+  geom_errorbarh(aes(xmin=lower, xmax=upper), height=0.2) +
+  geom_vline(xintercept=0, linetype="dashed", color="grey") +
+  theme_classic() +
+  labs(
+    x = "Observed Indirect Effect with Bootstrap CI",
+    y = "Mediator",
+    title = "Forest Plot of Mediation-specific Indirect Effects for Species"
+  ) +
+  scale_color_manual(values=c("black","red")) +
+  theme(legend.position="bottom")
+
+```
+
+Based on the forest plot results, we can see that the agathobaculum 
+butyriciproducens and escherichia coli pathways has significant indirect 
+mediation effects.
+
+Next we repeat the mediation analysis with the iHMP pathway abundances using the 
+same setup as before, using the time point as the treatment, dysbiosis score as 
+the outcome, but functional pathway abundance as the mediation variables instead 
```

Comment:
should we write "pathway abundance" instead of "functional pathway abundance"? Is there a relevant difference.